### ISSUE TITLE
[Installation] Installation breaks using Node >= 15

### ISSUE THREAD

**Issue by DyeffersonAz** [fomantic/Fomantic-UI#1859]
# Bug Report

I can't install it.

## Steps to reproduce
Look at the GIF (versions included):
![Peek 27-01-2021 10-21](https://user-images.githubusercontent.com/16638560/105996743-813d5b00-6089-11eb-81e2-f28533d796c3.gif)

Log attached.
[2021-01-27T13_20_17_699Z-debug.log](https://github.com/fomantic/Fomantic-UI/files/5880469/2021-01-27T13_20_17_699Z-debug.log)

**Comment by DyeffersonAz** [comment]
Sorry, I pressed enter too early. I'll edit it here. :sweat:

**Comment by lubber-de** [comment]
🤔 Could you try to run it using Node 12/13 ?

**Comment by Neo-Desktop** [comment]
I also have this issue with installing
I have the following, similar setup as @DyeffersonAz, just running on windows

```
>node --version
v15.7.0

>npm --version
7.4.3

>gulp --version
CLI version: 2.3.0
Local version: 4.0.2
```

It errors out in exactly the same way

[2021-01-28T11_10_35_678Z-debug.log](https://github.com/fomantic/Fomantic-UI/files/5886719/2021-01-28T11_10_35_678Z-debug.log)

**Comment by lubber-de** [comment]
@Neo-Desktop Again the question: Could you please try using Node 12/13 ?

**Comment by ko2in** [comment]
As far as I tested, it doesn't work with node version 15.0.0 and above. All prior versions (12, 13, 14) work.

![fui-npm](https://user-images.githubusercontent.com/930315/106264408-ff4d5d80-6253-11eb-8a7c-00968f23f226.gif)

**Comment by DyeffersonAz** [comment]
> @Neo-Desktop Again the question: Could you please try using Node 12/13 ?

As @ko2in tested, Nope. Same here

**Comment by lubber-de** [comment]
ok, but then it's a node 15 issue "only" . Current LTS of node is 14 which works, so we have to figure out what changed in node 15. Until then only node <= 14 is officially supported

**Comment by jeffrson** [comment]
I bet this is not related to Node 15, but npm7. And npm7 is official.

The interactive install breaks on all current package managers (yarn2, npm7, pnpm).

**Comment by lubber-de** [comment]
Closing, because i cannot reproduce this anymore using the latest LTS node versions node 12/14/16 and npm 6/8)
We changed our CI script to also test for Node 16 now which works

**Comment by jeffrson** [comment]
Very strange.
with npm 8.3.1 > `npm i --save-dev fomantic-ui`
```
[09:09:59] Starting 'run setup'...
npm ERR! ? Set-up Semantic UI (Use arrow keys)
npm ERR! ❯ Automatic (Use default locations and all components)
npm ERR!   Express (Set components and output folder)
npm ERR!   Custom (Customize all src/dist values) [09:09:59] The following tasks did not complete: install, run setup
npm ERR! [09:09:59] Did you forget to signal async completion?

npm ERR! A complete log of this run can be found in:
...
```

Same output (apart from "npm ERR!") with yarn 1.22.17. yarn3 simply puts
```
➤ YN0007: │ fomantic-ui@npm:2.8.8 must be built because it never has been before or the last one failed
➤ YN0009: │ fomantic-ui@npm:2.8.8 couldn't be built successfully (exit code 1, logs can be found here: /tmp/xfs-14304c61/build.log)
➤ YN0000: └ Completed in 4s 641ms
➤ YN0000: Failed with errors in 9s 988ms
```
Haven#t tried pnpm. But I guess it still works with npm6 only. So nothing changed at all :-(

**Comment by lubber-de** [comment]
🙄 Ok, i reopen again

**Comment by lubber-de** [comment]
I still cannot reproduce on node 15 and npm 7. It works for me without the "NPM ERR"
![image](https://user-images.githubusercontent.com/18379884/150822786-3dff8d92-1f1e-4dcf-a80b-3084b41c57eb.png)

I can confirm `yarn install` does not work, however `yarn gulp install` or `yarn gulp build` works as expected (afterwards) so `yarn install` does not start the npm install scripts. 

More investigation needed. Help is appreciated.

**Comment by lubber-de** [comment]
I think this is a general issue using inquirer in postinstall. At least when using yarn.
i found this issue https://github.com/yarnpkg/yarn/issues/7983 which describes the same situation but not using SUI/FUI

I updated inquirer to latest 8.2.0 by #2209 (should be included in tonights nightly build) which may have fixed it at least when using npm (that's perhaps the reason, why i cannot reproduce using npm on my machine). You may try again using npm tomorrow using the nightly npm build.

As said, when using `yarn gulp install` or `yarn gulp build` manually, it works, just not while doing automatic post install.

So i suggest to install via 

#### yarn 1.2
```terminal
yarn install --ignore-scripts
yarn gulp install
yarn gulp build
```

#### yarn 2.4
```terminal
yarn install --mode skip-build
yarn gulp install
yarn gulp build
```

**Comment by DifficultNick** [comment]
I'm sorry for repeated question, but is there any known way to use fomantic-ui for React now with node v16 and npm v8?

I got the same error as described above.

I tried steps from [comment in 1911](https://github.com/fomantic/Fomantic-UI/issues/1911#issuecomment-796848505) and got 'semantic' folder in my repo as supposed, but I'm not sure if I understand what to do with it.

P.S. I fell in love with your library at first sight, but still can't install it properly =(

**Comment by lubber-de** [comment]
I am sorry, I still cannot reproduce, please somebody else step in.
`npm install fomantic-ui --ignore-scripts` and `npx gulp build` works for me using node 12/14/16 and npm 6/7/8

React only needs the CSS, so if you only need the default theme, you might just install the precompiled version from https://github.com/fomantic/Fomantic-UI-CSS or https://github.com/fomantic/Fomantic-UI-LESS if you already have a LESS build somewhere in your project.

The only difference for 2.8.8 is LESS 4 and gulp-less 5 (which HAS to be used, otherwise it won't work with LESS 4.
Our nightly build uses node16/npm7

**Comment by DifficultNick** [comment]
Thanks for your answer!
Yeah, ignore-scripts + gulp install/build works. And now i realised why i was confused: I missed the thing that FUI (compared to SUI) doesn't have type definitions, so i thought it was incomplete result of compilation.

So let me ask you for advice (though it's not about this issue): if I want type definitions (because i use TS, not JS) should i use SUI and hope that FUI still will be merged back into it?

**Comment by lubber-de** [comment]
> if I want type definitions

You can still use the type definitions for SUI, however those won't cover the new features of FUI. If anybody volunteers to support/create/maintain type definitions for FUI, let us know!

> should i use SUI and hope that FUI still will be merged back into it?

Since quite some time, i believe, this hope will not be fullfilled....

**Comment by lubber-de** [comment]
Should now be fixed in #2272 because that wont start the interactive script automatically anymore
